doc/mclag/MCLAG_Enhancements_HLD.md

```diff
@@ -0,0 +1,810 @@
+
+Below Diagram show the remote ICCP MAC update to Kernel.
+
+![MCLAG ARP MAC kernel update Diagram](images/MclagArpFdbKernelUpdate.png "MCLAG ARP interested MAC kernel programming Flow")
+
+**Figure 2: MCLAG ARP interested MAC kernel programming Flow**
+
+## 3.5 SyncD
+No change to SyncD
+
+## 3.6 MclagSyncD
+### 3.6.1 FDB Changes
+MclagSyncd is extended to optimize the FDB events handling. 
+
+The changes include: -
+
+- sending local MAC updates to ICCPd from STATE_DB FDB_TABLE instead of ASIC_DB FDB_TABLE
+- sending of the MAC entries is event driven instead of the periodic 60 seconds polling
+- removal of the FDB cache in the MclagSyncd
+- adding the ICCP learned MAC addresses to APP_DB MCLAG_FDB_TABLE
+
+#### 3.6.1.1 Local MAC upstream using STATE_DB FDB_TABLE
+To distribute the local MAC addresses to ICCPd MclagSyncd will now listen for the STATE_DB FDB_TABLE updates. Current approach of periodic MAC table retrieval by polling and compare to find the MAC latest updates is costly in the scale scenarios. In the new approach STATE_DB FDB_TABLE MAC processing is event driven at MclagSyncd and no periodic polling is done.
+
+#### 3.6.1.2 Removal of FDB table in MclagSyncd
+With the above change to event-driven updates, MclagSyncd no longer needs a local FDB table copy - it can update directly between ICCPd and FdbOrch.
+
+#### 3.6.1.3 Populating ICCP MAC updates to APP_DB MCLAG_FDB_TABLE
+
+In current implementation MclagSyncd updates the MAC addresses into the existing APP_DB FDB_TABLE. In this enhancement, a new APP_DB MCLAG_FBD_TABLE is used. This allows clients to distinguish between local and remote (ICCP received) MAC address.
+
+#### 3.6.1.4 Populating ICCP and MCLAG interface state information
+
+MclagSyncd is extended to receive the ICCP session related information from ICCPd and to populate this information in STATE_DB STATE_MCLAG_TABLE. MclagSyncd also receives MCLAG remote interface information which is populated in STATE_DB STATE_MCLAG_REMOTE_INTF_TABLE.
+
+### 3.6.2 MCLAG config handling
+MclagSyncd will be enhanced to listen to CONFIG_DB changes in MCLAG domain table and MCLAG_INTF_TABLE and notify ICCPd. ICCPd is enhanced to process the configuration and handle config changes accordingly.
+
+## 3.7 SAI
+### 3.7.1 Port Isolation
+The following SAI definitions will be used and no enhancements necessary
+
+- https://github.com/opencomputeproject/SAI/blob/master/inc/saiisolationgroup.h
+
```

Comment:
Please also add details about the SAI_FDB_ENTRY_TYPE_STATIC_MACMOVE. As of today, this enum does not exist in SAI.

https://github.com/opencomputeproject/SAI/blob/master/inc/saifdb.h#L47

Reply:
Thanks for the comment. SAI changes will be added as part of common SAI PR. will add details here once available.